extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java

```diff
@@ -332,4 +340,14 @@ HealthBuildItem addHealthCheck(MongoClientBuildTimeConfig buildTimeConfig) {
         return new HealthBuildItem("io.quarkus.mongodb.health.MongoHealthCheck",
                 buildTimeConfig.healthEnabled, "mongodb");
     }
+
+    @BuildStep
+    void produceMongoClientSettings(
```

Comment:
We usually name a build step method by what it does. Here you setup metrics or you add the listener so maybe `setupMetrics` or `addConnectionPoolListener` are better names.